packages/shared/babel-ast-utils/src/generator.js

```diff
@@ -0,0 +1,225 @@
+  },
+  BooleanLiteral(node, state) {
+    node.type = 'Literal';
+    this.Literal(node, state);
+  },
+  NullLiteral(node, state) {
+    node.type = 'Literal';
+    node.raw = 'null';
+    node.value = null;
+    this.Literal(node, state);
+  },
+  RegExpLiteral(node, state) {
+    node.type = 'Literal';
+    node.raw = node.extra.raw;
+    node.value = {};
+    node.regex = {
+      pattern: node.pattern,
+      flags: node.flags,
+    };
+    this.Literal(node, state);
+  },
+  BigIntLiteral(node, state) {
+    node.type = 'Literal';
+    node.raw = node.extra.raw;
+    this.Literal(node, state);
+  },
+  ObjectProperty(node, state) {
+    node.type = 'Property';
+    node.kind = 'init';
+    if (
+      (node.shorthand && node.value.type !== 'Identifier') ||
+      node.value.name !== node.key.name
+    ) {
+      node.shorthand = false;
+    }
+    this.Property(node, state);
+  },
+  ObjectMethod(node, state) {
+    node.value = {
+      type: 'FunctionExpression',
+      id: node.id,
+      params: node.params,
+      body: node.body,
+      async: node.async,
+      generator: node.generator,
+      expression: node.expression,
+    };
+
+    node.type = 'Property';
+    if (node.kind === 'method') {
+      node.kind = 'init';
+    }
+
+    this.Property(node, state);
+  },
+  ClassMethod(node, state) {
+    node.value = {
+      type: 'FunctionExpression',
+      id: node.id,
+      params: node.params,
+      body: node.body,
+      async: node.async,
+      generator: node.generator,
+      expression: node.expression,
+    };
+
+    node.type = 'MethodDefinition';
+    this.MethodDefinition(node, state);
+  },
+  ClassPrivateMethod(node, state) {
+    node.value = {
+      type: 'FunctionExpression',
+      id: node.id,
+      params: node.params,
+      body: node.body,
+      async: node.async,
+      generator: node.generator,
+      expression: node.expression,
+    };
+
+    node.type = 'MethodDefinition';
+    this.MethodDefinition(node, state);
+  },
+  ClassProperty(node, state) {
+    if (node.static) {
+      state.write('static ');
+    }
+
+    if (node.computed) {
+      state.write('[');
+      this[node.key.type](node.key, state);
+      state.write(']');
+    } else {
+      this[node.key.type](node.key, state);
+    }
+
+    if (node.value) {
+      state.write(' = ');
+      this[node.value.type](node.value, state);
+    }
+
+    state.write(';');
+  },
+  ClassPrivateProperty(node, state) {
+    if (node.static) {
+      state.write('static ');
+    }
+
+    this[node.key.type](node.key, state);
+    if (node.value) {
+      state.write(' = ');
+      this[node.value.type](node.value, state);
+    }
+
+    state.write(';');
+  },
+  PrivateName(node, state) {
+    state.write('#' + node.name, node);
+  },
+  Import(node, state) {
+    // astring doesn't support ImportExpression yet
+    state.write('import');
+  },
+  // TODO: OptionalMemberExpression, OptionalCallExpression once astring supports ChainExpression
```

Comment:
We can probably put this in our fork as well?

https://github.com/davidbonnet/astring/pull/424

Not sure if the test in the PR shouldn't actually just be a fixture instead, seems easier.

Reply:
done!